cfsdk/src/main/java/io/harness/cfsdk/CfClient.java

```diff
@@ -121,6 +122,9 @@ private void initializeInternal(final Context context, final String apiKey, fina
 
         } catch (Exception e) {
             log.warn(e.getMessage(), e);
+            if (authCallback != null) {
```

Comment:
Unrelated to the main issue with closing the SDK client. But getting it fixed in this PR.